akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -1776,7 +1776,9 @@ trait FlowOps[+Out, +Mat] {
    * See also [[FlowOps.conflate]], [[FlowOps.limit]], [[FlowOps.limitWeighted]] [[FlowOps.batch]] [[FlowOps.batchWeighted]]
    */
   def conflateWithSeed[S](seed: Out => S)(aggregate: (S, Out) => S): Repr[S] =
-    via(Batch(1L, ConstantFun.zeroLong, seed, aggregate).withAttributes(DefaultAttributes.conflate))
+    via(
+      Batch(1L, ConstantFun.zeroLong, seed, aggregate)
+        .withAttributes(DefaultAttributes.conflate and SourceLocation.forLambda(aggregate)))
```

Comment:
how does the SourceLocation cache work? The key is the class of the lambda. Can't this be used with a `scala.Function2`?

Reply:
I didn't think much about the cache lookup, just copied it from the original PR, maybe that can be a problem, I'll look into that. When benchmarking it seemed the cache does not do a huge difference for performance so maybe we should just skip it instead.

Re `Function2` that works:

```
scala> akka.stream.Attributes.SourceLocation.forLambda(akka.stream.scaladsl.Keep.left)
res1: akka.stream.Attributes.SourceLocation = SourceLocation(akka.stream.scaladsl-Materialization.scala:14)
```

Reply:
That is still a lambda class. I was thinking the abnormal usage of `new Function2` from several different places. Maybe that doesn't have a source location anyway.

Reply:
I think it works:

```
scala> val f = new Function2[Int, Int, Int] {
     | def apply(a: Int, b: Int): Int = a + b
     | }
f: (Int, Int) => Int = <function2>

scala> f.getClass
res0: Class[_ <: (Int, Int) => Int] = class $anon$1

scala> akka.stream.Attributes.SourceLocation.forLambda(f)
res1: akka.stream.Attributes.SourceLocation = $line6-<console>:26
```

It still gets a unique class because it is an anonymous class, however I removed the caching now so it does not matter. :)